source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -70,7 +77,12 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap&, bool) {
 
   config->stats().enabled_.inc();
 
-  if (config->requestAllowed()) {
+  std::vector<RateLimit::LocalDescriptor> descriptors;
```

Comment:
For perf reasons I wonder if you want to use InlinedVector here instead? I'm guessing in general the number of descriptors will be small?

Reply:
Yeah, makes sense. Although it's a vector of vectors, so savings are not applying to inner vectors.

Reply:
Tried that but it seems to leak quite a bit deep into envoy. Is there some writable interface that we could use like Span?

Reply:
Let's leave it for later? We'd need to change all rate limit "populate" interfaces to take inlined vector I think to make it work.

Reply:
That's fine don't worry about it.